test/edu/washington/escience/myria/operator/SplitTest.java

```diff
@@ -0,0 +1,167 @@
+    Split splitOp = new Split(new TupleSource(input), 0, ":");
+
+    splitOp.open(TestEnvVars.get());
+    TupleBatch result;
+    int rowIdx = 0;
+    while (!splitOp.eos()) {
+      result = splitOp.nextReady();
+      if (result != null) {
+        assertEquals(schema.numColumns(), result.getSchema().numColumns());
+        assertEquals(Type.STRING_TYPE, result.getSchema().getColumnType(0));
+
+        for (int batchIdx = 0; batchIdx < result.numTuples(); ++batchIdx, ++rowIdx) {
+          assertEquals(expectedResults[rowIdx], result.getString(0, batchIdx));
+        }
+      }
+    }
+    assertEquals(expectedResults.length, rowIdx);
+    splitOp.close();
+  }
+
+  /**
+   * Test output spanning multiple batches. All integers from 0 to 2 * TupleBatch.BATCH_SIZE + 1 are
+   * concatenated as comma-separated strings in rows of 10 each. Result should contain each integer
+   * from the input in its own row.
+   * 
+   * @throws DbException
+   */
+  @Test
+  public void testAllBatchesReturned() throws DbException {
+
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.STRING_TYPE), ImmutableList.of("joined_ints"));
+    final TupleBatchBuffer input = new TupleBatchBuffer(schema);
+    final long expectedResults = 2 * TupleBatch.BATCH_SIZE + 1;
+    final int intsInRow = 10;
+    for (long i = 0; i < (expectedResults / intsInRow) + 1; ++i) {
+      final int remainder = (int) (expectedResults % intsInRow);
+      final int intsToConcatSize;
+      if ((expectedResults - (i * intsInRow)) == remainder) {
+        intsToConcatSize = remainder;
+      } else {
+        intsToConcatSize = intsInRow;
+      }
+      final long[] intsToConcat = new long[intsToConcatSize];
+      for (int j = 0; j < intsToConcatSize; ++j) {
+        intsToConcat[j] = (i * intsInRow) + j;
+      }
+      input.putString(0, Longs.join(",", intsToConcat));
+    }
+
+    Split splitOp = new Split(new TupleSource(input), 0, ",");
+    splitOp.open(TestEnvVars.get());
+    TupleBatch result;
+    long rowIdx = 0;
+    while (!splitOp.eos()) {
+      result = splitOp.nextReady();
+      if (result != null) {
+        assertEquals(schema.numColumns(), result.getSchema().numColumns());
+        assertEquals(Type.STRING_TYPE, result.getSchema().getColumnType(0));
+
+        for (int batchIdx = 0; batchIdx < result.numTuples(); ++batchIdx, ++rowIdx) {
+          assertEquals(rowIdx, Long.parseLong(result.getString(0, batchIdx)));
+        }
+      }
+    }
+    assertEquals(expectedResults, rowIdx);
+    splitOp.close();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testSplitColumnInvalidType() throws DbException {
+    final Schema schema = new Schema(ImmutableList.of(Type.LONG_TYPE), ImmutableList.of("long"));
+    final TupleBatchBuffer input = new TupleBatchBuffer(schema);
+    input.putLong(0, 1L);
+    Split splitOp = new Split(new TupleSource(input), 0, ":");
+    splitOp.open(TestEnvVars.get());
+    splitOp.close();
```

Comment:
When I did this, I assumed (perhaps erroneously) that the line I expect to throw the exception should be the last line of the test. That way if something goes wrong with the test I find out.

This is extraordinarily unlikely, but I wouldn't want the test to "pass" because `close` throws the exception.

IMO style and symmetry aren't so important for tests, so much as conciseness/understandability and most importantly, correctness.